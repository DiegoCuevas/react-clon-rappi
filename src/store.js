import { createStore, applyMiddleware, compose } from "redux";
import reducer from "./reducer";
import thunk from "redux-thunk";
import logger from "redux-logger";
import offlineSync from "./midlewares/offline-sync";

const composer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

if (localStorage.getItem("version") !== process.env.REACT_APP_VERSION) {
  localStorage.removeItem("state");
}
const preloadedState = JSON.parse(localStorage.getItem("state") || "{}");

const store = createStore(
  reducer,
  // preloadedState,
  // composer(applyMiddleware(thunk, logger, offlineSync))
  composer(applyMiddleware(thunk, logger))
);

export default store;
