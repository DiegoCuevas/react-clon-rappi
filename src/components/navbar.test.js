import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";

import store from "../store";

import Navbar from "./navbar";

test("Navbar component", () => {
  const { asFragment } = render(
    <Provider store={store}>
      <Navbar />
    </Provider>
  );

  expect(asFragment()).toMatchSnapshot();
});
