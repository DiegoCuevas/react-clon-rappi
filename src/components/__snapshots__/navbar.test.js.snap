// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component 1`] = `
<DocumentFragment>
  .emotion-9 {
  border-bottom: 1px solid rgba(0,0,0,.0975);
  margin-bottom: 2em;
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-item: center;
  -webkit-box-align-item: center;
  -ms-flex-align-item: center;
  align-item: center;
  margin: 0 auto;
  padding: .5em 1.5em;
  list-style: none;
  color: #fff;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-1 {
  margin-right: 1em;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #000;
}

.emotion-1:hover {
  color: #003569;
}

.emotion-0 {
  width: 80px;
}

.emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100px;
}

<nav
    class="emotion-9"
  >
    <ul
      class="emotion-8"
    >
      <li
        class="emotion-2"
      >
        <a
          aria-current="page"
          class="emotion-1"
          href="/"
        >
          <img
            alt="rappi"
            class="emotion-0"
            src="https://images.rappi.com/web/logo-rappi-new.svg?e=webp"
          />
        </a>
      </li>
      <div
        class="emotion-7"
      >
        <li>
          <a
            class="emotion-1"
            href="/cart"
          >
            <span
              class="emotion-2"
            >
              Cart
            </span>
          </a>
        </li>
        <li>
          <a
            class="emotion-1"
            href="/login"
          >
            <span
              class="emotion-2"
            >
              Logout
            </span>
          </a>
        </li>
      </div>
    </ul>
  </nav>
</DocumentFragment>
`;
